extensions/ql-vscode/src/view/results/alert-table-header.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from "react";
+import { useCallback } from "react";
+import { vscode } from "../vscode-api";
+import {
+  InterpretedResultsSortColumn,
+  InterpretedResultsSortState,
+  SortDirection,
+} from "../../common/interface-types";
+import { nextSortDirection } from "./result-table-utils";
+
+export function AlertTableHeader({
+  sortState,
+}: {
+  sortState?: InterpretedResultsSortState;
+}) {
+  const sortClass = useCallback(
+    (column: InterpretedResultsSortColumn): string => {
+      if (sortState !== undefined && sortState.sortBy === column) {
+        return sortState.sortDirection === SortDirection.asc
+          ? "sort-asc"
+          : "sort-desc";
+      } else {
+        return "sort-none";
+      }
+    },
+    [sortState],
+  );
+
+  const getNextSortState = useCallback(
+    (
+      column: InterpretedResultsSortColumn,
+    ): InterpretedResultsSortState | undefined => {
+      const prevDirection =
+        sortState && sortState.sortBy === column
+          ? sortState.sortDirection
+          : undefined;
+      const nextDirection = nextSortDirection(prevDirection, true);
+      return nextDirection === undefined
+        ? undefined
+        : { sortBy: column, sortDirection: nextDirection };
+    },
+    [sortState],
+  );
+
+  const toggleSortStateForColumn = useCallback(
+    (column: InterpretedResultsSortColumn): void => {
+      vscode.postMessage({
+        t: "changeInterpretedSort",
+        sortState: getNextSortState(column),
+      });
+    },
+    [getNextSortState],
+  );
+
+  const clickCallback = useCallback(
```

Comment:
I think a name like `handleClick` would be better, to match the `onClick` prop.

Reply:
Actually, now that `toggleSortStateForColumn` doesn't take any arguments, I think we can safely delete the `clickCallback`/`handleClick` method. We can instead pass in `toggleSortStateForColumn` directly to the `onClick` prop.